react/components/__tests__/QuantitySelector.test.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react'
+import { render, screen, fireEvent } from '@vtex/test-tools/react'
+import { ToastContext } from '@vtex/styleguide/lib/ToastProvider'
+
+import QuantitySelector from '../QuantitySelector'
+
+describe('<QuantitySelector />', () => {
+  it('should change from dropdown to input after quantity exceeds max value', () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector id="1" value={5} maxValue={50} onChange={onChange} />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toBeInTheDocument()
+
+    expect(dropdown).toHaveValue('5')
+
+    fireEvent.change(dropdown, { target: { value: '10' } })
+
+    expect(onChange).toHaveBeenLastCalledWith(10)
+    expect(dropdown).not.toBeInTheDocument()
+
+    const [input] = screen.getAllByRole('textbox')
+
+    expect(input).toHaveValue('10')
+  })
+
+  it('should change back to input after changing the value to under max value', () => {
+    const onChange = jest.fn()
+
+    render(
+      <QuantitySelector id="1" value={10} maxValue={50} onChange={onChange} />
+    )
+
+    const [input] = screen.getAllByRole('textbox')
+
+    expect(input).toBeInTheDocument()
+
+    fireEvent.change(input, { target: { value: '5' } })
+    fireEvent.blur(input)
+
+    expect(onChange).toHaveBeenLastCalledWith(5)
+    expect(input).not.toBeInTheDocument()
+  })
+
+  it("shouldn't display default measurement unit value", () => {
+    render(
+      <QuantitySelector
+        id="1"
+        value={1}
+        maxValue={50}
+        onChange={jest.fn()}
+        measurementUnit="un"
+      />
+    )
+
+    const [dropdown] = screen.getAllByRole('combobox')
+
+    expect(dropdown).toHaveDisplayValue('1')
```

Comment:
Why isn't `expect(dropdown).toBeInTheDocument()` present here? Even though it's present at least once on previous tests, shouldn't each test be unique? Or that would just add unnecessary complexity?

Reply:
(same goes for the next few tests)

Reply:
I think it adds unnecessary complexity, else we would need to check for everything the other tests also check on this one... I like to keep the tests short and to test only one thing (or one "flow") so when when one of them fails it is clear what has been broken

Reply:
I see, my concern was in a scenario which the prop values would break the component, so the previous tests would pass but not this one. But sure, if that happens, I don't think it would be hard to catch.

Reply:
hmmm, I can add this one _little_ check for this one though, not completely against it

Reply:
oh wait, I think it the document doesn't have the thing I'm trying to get using `getAllByRole` or `getByRole`, testing-library will already throw an exception for me, so I think the `toBeInTheDocument()` call is actually unnecessary

Reply:
[bingo](https://testing-library.com/docs/dom-testing-library/api-queries#getallby)

Reply:
Ohh, I just assumed `getAllByRole` would return an empty array on 0 matches, cool

Reply:
In that case, could you remove the previous `toBeInTheDocument()` assertions? Lines 17 and 40.

Reply:
sure!

Reply:
done